web/packages/teleport/src/UnifiedResources/Resources.tsx

```diff
@@ -53,19 +66,120 @@ const FETCH_MORE_SIZE = 24;
 
 const loadingCardArray = new Array(FETCH_MORE_SIZE).fill(undefined);
 
+export const PINNING_NOT_SUPPORTED_MESSAGE =
+  'This cluster does not support pinning resources. To enabled, upgrade to 14.1 or newer.';
+
+const tabs: { label: string; value: UnifiedTabPreference }[] = [
+  {
+    label: 'All Resources',
+    value: UnifiedTabPreference.All,
+  },
+  {
+    label: 'Pinned Resources',
+    value: UnifiedTabPreference.Pinned,
+  },
+];
+
 export function Resources() {
-  const { isLeafCluster } = useStickyClusterId();
+  const { isLeafCluster, clusterId } = useStickyClusterId();
   const enabled = localStorage.areUnifiedResourcesEnabled();
+  const pinningNotSupported = localStorage.arePinnedResourcesDisabled();
+  const {
+    getClusterPinnedResources,
+    preferences,
+    updatePreferences,
+    updateClusterPinnedResources,
+  } = useUser();
   const teleCtx = useTeleport();
   const canCreate = teleCtx.storeUser.getTokenAccess().create;
-  const { clusterId } = useStickyClusterId();
+  const [selectedResources, setSelectedResources] = useState<string[]>([]);
+
+  const [getPinnedResourcesAttempt, getPinnedResources, setPinnedResources] =
+    useAsync(
+      useCallback(
+        () => getClusterPinnedResources(clusterId),
+        [clusterId, getClusterPinnedResources]
+      )
+    );
+
+  useEffect(() => {
+    getPinnedResources();
```

Comment:
We should also clean up the "update" attempt, so if you had an error on a cluster A and you switch to a cluster B you won't see the old error.

```
useEffect(() => {
  setUpdatePinnedResources(makeEmptyAttempt());
  getPinnedResources();
}, [clusterId, getPinnedResources, setUpdatePinnedResources]);
```

BTW I'm going to refactor it; when changing the cluster, the whole `Resources` component will be re-mounted and this cleanup logic will no longer be needed.